hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieFileGroupReaderBasedFileFormat.scala

```diff
@@ -202,6 +214,8 @@ class HoodieFileGroupReaderBasedFileFormat(tablePath: String,
     val requestedAvroSchema = AvroSchemaUtils.pruneDataSchema(avroTableSchema, AvroConversionUtils.convertStructTypeToAvroSchema(requestedSchema, sanitizedTableName), exclusionFields)
     val dataAvroSchema = AvroSchemaUtils.pruneDataSchema(avroTableSchema, AvroConversionUtils.convertStructTypeToAvroSchema(dataSchema, sanitizedTableName), exclusionFields)
 
+    spark.sessionState.conf.setConfString("spark.sql.parquet.enableVectorizedReader", supportVectorizedRead.toString)
```

Comment:
A friendly reminder: 
If we modify this configuration in the conf of spark sessionState in the hudi logic, it may disrupt the read logic of other datasources.
For example, if this configuration is initially set to true, When a spark sql reads a hudi table and another datasource table such as a hive table, the behavior we hope for is that whether the hudi performs vectorized reading is controlled by the hudi logic itself, while hive directly performs vectorized reading.
However, if we change this configuration here, perhaps this will lead to hive not performing vectorized reading.
A similar issue was encountered in the previous `BaseRelation.java`.
cc @jonvex @yihua

Reply:
you mean the override of the conf in spark session would affect the behavior of federation queries with multiple data sources in one spark workload, like Hive data source together with Hudi?

@yihua it looks like a regression though.

Reply:
yes